model/src/main/kotlin/HashAlgorithm.kt

```diff
@@ -122,21 +123,37 @@ enum class HashAlgorithm(private vararg val aliases: String, val verifiable: Boo
     /**
      * Return the hexadecimal digest of this hash for the given [file].
      */
-    fun calculate(file: File): String =
-        file.inputStream().use { inputStream ->
-            // 4MB has been chosen rather arbitrarily, hoping that it provides good performance while not consuming a
-            // lot of memory at the same time, also considering that this function could potentially be run on multiple
-            // threads in parallel.
-            val buffer = ByteArray(4 * 1024 * 1024)
-            val digest = getMessageDigest(file)
-
-            var length: Int
-            while (inputStream.read(buffer).also { length = it } > 0) {
-                digest.update(buffer, 0, length)
-            }
+    fun calculate(file: File): String = file.inputStream().use { calculate(it, file.length()) }
+
+    /**
+     * Return the hexadecimal digest of this hash for the given [resourceName].
+     */
+    fun calculate(resourceName: String): String {
+        val resource = javaClass.getResource(resourceName)
```

Comment:
Shouldn't we run into the same problems here as described in my Slack message and here: https://stackoverflow.com/questions/10144210/java-jar-file-use-resource-errors-uri-is-not-hierarchical? 

Is there a test to this?

Reply:
No, we should not run into the same issue, as we're not converting a `URL` / `URI` to a `File` here. But I can look into adding a test.